src/app/dataset/datasetlog/datasetlog.component.ts

```diff
@@ -71,19 +78,38 @@ export class DatasetlogComponent implements OnInit {
   /* get content of log, based on external taskid and topology
   */
   returnLog() {
+
     if (!this.isShowingLog || !this.isShowingLog['externaltaskId'] || !this.isShowingLog['topology']) { return false; }
-    this.logPlugin = this.isShowingLog['plugin']
+      
+    if (this.logPlugin !== this.isShowingLog['plugin'] && this.isShowingLog['processed'] === 0) {
+      this.logFrom = 1;
+      this.logTo = this.logPerStep;
+      this.logStep = 1;
+    }
+
+    this.logPlugin = this.isShowingLog['plugin'];
+
+    if (this.isShowingLog['processed'] && (this.isShowingLog['status'] === 'FINISHED') || (this.isShowingLog['status'] === 'CANCELLED') || (this.isShowingLog['status'] === 'FAILED')) { 
+      this.logTo = this.isShowingLog['processed']; 
+      this.logFrom = (this.logTo - this.logPerStep) >= 1 ? (this.logTo - this.logPerStep) : 1;
```

Comment:
Did you forget the +1 like you have on line 52?
This suggests maybe create a small function for calculating that logFrom if it should be identical.